server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+        mock_repos = [{'id': 'mock1'}, {'id': 'mock2'}]
+        repos_view = ReposView()
+        repos_view._process_repos(mock_repos, True, True)
+        mock_merge.assert_has_calls(
+            [mock.call('importers', mock_factory.repo_importer_manager(), mock_repos),
+             mock.call('distributors', mock_factory.repo_distributor_manager(), mock_repos)]
+        )
+        mock_convert_date.assert_has_calls(
+            [mock.call({'id': 'mock1', '_href': '/mock/path/'}),
+             mock.call({'id': 'mock2', '_href': '/mock/path/'})]
+        )
+
+    @mock.patch('pulp.server.webservices.views.repositories._convert_repo_dates_to_strings')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    def test__process_repos_scratchpad(self, mock_reverse, mock_convert_date):
+        """
+        Test that scratchpad field has been removed.
+        """
+
+        mock_reverse.return_value = '/mock/path/'
+        mock_repos = [{'id': 'mock1', 'scratchpad': 'should be removed'}, {'id': 'mock2'}]
+        repos_view = ReposView()
+        repos_view._process_repos(mock_repos, False, False)
+        mock_convert_date.assert_has_calls(
+            [mock.call({'id': 'mock1', '_href': '/mock/path/'}),
+             mock.call({'id': 'mock2', '_href': '/mock/path/'})]
+        )
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch('pulp.server.webservices.views.repositories.ReposView._process_repos')
+    @mock.patch('pulp.server.webservices.views.repositories.RepoModel.get_collection')
+    def test_get_repos_no_options(self, mock_collection, mock_process):
+        """
+        Get repos without passing options.
+        """
+
+        mock_repos = [{'mock_repo_1': 'somedata'}, {'mock_repo_2': 'moredata'}]
+        mock_collection.return_value.find.return_value = mock_repos
+        mock_request = mock.MagicMock()
+        mock_request.GET = {}
+        repos_view = ReposView()
+
+        repos_view.get(mock_request)
+        mock_process.assert_called_once_with(mock_repos, False, False)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch('pulp.server.webservices.views.repositories.ReposView._process_repos')
+    @mock.patch('pulp.server.webservices.views.repositories.RepoModel.get_collection')
+    def test_get_repos_with_details(self, mock_collection, mock_process):
+        """
+        Get repos with the details shortcut.
+        """
+
+        mock_repos = [{'mock_repo_1': 'somedata'}, {'mock_repo_2': 'moredata'}]
+        mock_collection.return_value.find.return_value = mock_repos
+        mock_request = mock.MagicMock()
+        mock_request.GET = {'details': 'true'}
+        repos_view = ReposView()
+
+        repos_view.get(mock_request)
+        mock_process.assert_called_once_with(mock_repos, True, True)
```

Comment:
since the only difference is the value of `include_importers` and `include_distributors`, I think the response body is not necessary.